### PR TITLE
[acls] bug fixes and features

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY
ACL bug fixes and features

- _eq_ commands now are interpreted correctly for comparison.  `20 permit udp any any eq 23`  - > `20 permit udp any any eq telnet`
- for _replaced_ and _overridden_ state negate commands are fired before normal commands.
- added support for `object-group-service` under the protocol attribute

##### ISSUE TYPE

- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME

ios_acls

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by ashwini-mhatre — COMMENTED** [review]
could you please add some tests

File: plugins/module_utils/network/ios/config/acls/acls.py [hunk 1]
```diff
@@ -101,21 +101,35 @@ def _compare(self, want, have, afi):
...
+            begin = len(self.commands)  # to determine the index for acl command
             self._compare_aces(
                 wentry.pop("aces", {}),
                 hentry.pop("aces", {}),
                 afi,
                 wname,
             )  # handle aces
 
+            self.commands[begin : len(self.commands)] = rearrange_cmds(
+                self.commands[begin : len(self.commands)],
```
Comment (Qalthos) [inline plugins/module_utils/network/ios/config/acls/acls.py:130]: I think this is the solution to the sanity issue.
```suggestion
            self.commands[begin:len(self.commands)] = rearrange_cmds(
                self.commands[begin:len(self.commands)],
```

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: plugins/module_utils/network/ios/config/acls/acls.py (hunk 1 above)
Comment (Qalthos) [reply plugins/module_utils/network/ios/config/acls/acls.py:130]: Ugh yeah, black is the one introducing this change. We will have to do something about that.

Comment (Qalthos) [reply plugins/module_utils/network/ios/config/acls/acls.py:130]: Is `len(self.commands)` meant to change over the course of execution? Can we not instead do something like this?
```suggestion
            end = len(self.commands)
            self.commands[begin:end] = rearrange_cmds(
                self.commands[begin:],
```

**Review by rohitthakur2590 — APPROVED** [review]

**Pull request merged**